main.go

```diff
@@ -138,11 +152,53 @@ func walk(ch chan<- *file, start string) {
 }
 
 func addLicense(path string, fmode os.FileMode, tmpl *template.Template, data *copyrightData) (bool, error) {
+	var lic []byte
+	var err error
+	lic, err = getLicenseForExtension(path, tmpl, data)
+	if err != nil || lic == nil {
+		return false, err
+	}
+
+	b, err := ioutil.ReadFile(path)
+	if err != nil || hasLicense(b) {
+		return false, err
+	}
+
+	line := hashBang(b)
+	if len(line) > 0 {
+		b = b[len(line):]
+		if line[len(line)-1] != '\n' {
+			line = append(line, '\n')
+		}
+		lic = append(line, lic...)
+	}
+	b = append(lic, b...)
+	return true, ioutil.WriteFile(path, b, fmode)
+}
+
+func checkLicense(path string, tmpl *template.Template, data *copyrightData) (bool, error) {
+	var lic []byte
+	var err error
+	lic, err = getLicenseForExtension(path, tmpl, data)
+
+	if err != nil || lic == nil {
+		return false, err
+	}
+
+	b, err := ioutil.ReadFile(path)
+	if err != nil || hasLicense(b) {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func getLicenseForExtension(path string, tmpl *template.Template, data *copyrightData) ([]byte, error) {
```

Comment:
Go discourages "get" in the naming (https://golang.org/doc/effective_go.html#Getters) plus the function doesn't take an extension. I'd rename it to `licenseHeader`.